query/src/sql/planner/binder/select.rs

```diff
@@ -56,7 +57,7 @@ pub struct SelectItem<'a> {
 impl<'a> Binder {
     pub(super) async fn bind_select_stmt(
         &mut self,
-        bind_context: &BindContext,
+        bind_context: &mut BindContext,
```

Comment:
Seems there is no need to make this mutable. Since we will always return an owned `BindContext`.

Reply:
`bind_context` will be used in `bind_table_reference` which requires it mutable.